field_utils/mapl_udunits2cfunc.c

```diff
@@ -0,0 +1,16 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include "udunits2.h"
+
+/* Helper function to augment udunits2 error handling
+ * Sets the udunits2 error handler to ut_ignore
+ * which disables error messages from udunits2
```

Comment:
Q:   If this is just disabling error messages, do we instead want to pass a procedure that directs error messages to MAPL's error handling routine?

If it is also disabling "info" messages, then, yes, we want to shut it all down.

Reply:
It includes informational/warning messages. udunits sends messages to the effect of "Unit A is being used instead of Unit B" for several units to the error handler each time the units database, which currently is 1 time per run. There are also status codes reporting success, failure, or information. So it's not an error. To avoid the helper function, we could allow the messages since it only happens once per run. I don't like that from a design point of view, but practicality may win out here.

Reply:
No - I don't want any output to STDOUT if things are "good".

Reply:
Resolving for now to allow merge.

Reply:
I know this is merged.

For completeness: the error message handler writes 9 long lines to STDERR when it starts up. Each line reports an override like this:

> Definition of "au" in "/discover/swdev/gmao_SIteam/Baselibs/ESMA-Baselibs-7.17.0/x86_64-pc-linux-gnu/gfortran_12.1.0-openmpi_4.1.3-SLES12/Linux/share/udunits/udunits2-accepted.xml", line 123, overrides prefixed-unit "1.6605402e-45 kilogram"

So it is the normal course. We haven't made any changes to the units.

So udunits2 produces messages that are sent to the ut_error_message_handler, which defaults to STDERR, if the message is not an error. We can change the ut_error_message_handler to ut_ignore (or another handler), but it has to be done in C since the handler has variadic arguments which can't be interfaced from Fortran.